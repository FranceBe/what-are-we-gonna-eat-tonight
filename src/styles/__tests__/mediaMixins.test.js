import {
  mediaMin,
  mediaMax,
  mediaOnly,
} from 'styles/mediaMixins'

describe('Media style mixin', () => {
  describe('mediaMin', () => {
    describe('Given mediaMin is called on existing size', () => {
      it('Should return min media query and content', () => {
        expect(mediaMin.sm('test')).toMatchSnapshot()
      })
    })
  })

  describe('mediaMax', () => {
    describe('Given mediaMax is called on existing size', () => {
      it('Should return max media query and content', () => {
        expect(mediaMax.sm('test')).toMatchSnapshot()
      })
    })
  })

  describe('mediaOnly', () => {
    describe('Given mediaOnly is called on existing min and max size', () => {
      it('Should return min and max media query and content', () => {
        expect(mediaOnly.sm('test')).toMatchSnapshot()
      })
    })

    describe('Given mediaOnly is called on existing max size', () => {
      it('Should return max media query and content', () => {
        expect(mediaOnly.xs('test')).toMatchSnapshot()
      })
    })

    describe('Given mediaOnly is called on existing min size', () => {
      it('Should return min media query and content', () => {
        expect(mediaOnly.xl('test')).toMatchSnapshot()
      })
    })
  })
})
