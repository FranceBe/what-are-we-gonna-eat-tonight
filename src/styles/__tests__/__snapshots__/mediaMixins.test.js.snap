// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Media style mixin mediaMax Given mediaMax is called on existing size Should return max media query and content 1`] = `
Array [
  "@media (max-width:",
  "47.9375",
  "em){",
  "test",
  "}",
]
`;

exports[`Media style mixin mediaMin Given mediaMin is called on existing size Should return min media query and content 1`] = `
Array [
  "@media (min-width:",
  "36",
  "em){",
  "test",
  "}",
]
`;

exports[`Media style mixin mediaOnly Given mediaOnly is called on existing max size Should return max media query and content 1`] = `
Array [
  "@media ",
  "(max-width: 47.9375em)",
  "{",
  "test",
  "}",
]
`;

exports[`Media style mixin mediaOnly Given mediaOnly is called on existing min and max size Should return min and max media query and content 1`] = `
Array [
  "@media ",
  "(min-width: 36em)",
  " and ",
  "(max-width: 61.9375em)",
  "{",
  "test",
  "}",
]
`;

exports[`Media style mixin mediaOnly Given mediaOnly is called on existing min size Should return min media query and content 1`] = `
Array [
  "@media ",
  "(min-width: 75em)",
  "{",
  "test",
  "}",
]
`;
